src/JLD.jl

```diff
@@ -1209,7 +1209,7 @@ FileIO.save(f::File{format"JLD"}, value...; kwargs...) = error("Must supply a na
 # load with just a filename returns a dictionary containing all the variables
 function FileIO.load(f::File{format"JLD"})
     jldopen(FileIO.filename(f), "r") do file
-        (String => Any)[var => read(file, var) for var in names(file)]
+        Dict{String,Any}(var => read(file, var) for var in names(file))
```

Comment:
Unfortunately there is no way to fix that deprecation because it is
unparsable in Julia 4. So we will have to live with it until a Julia
.5-only fork of this package is made.
On Mon, Jun 13, 2016 at 8:54 AM Abhijith Chandraprabhu <
notifications@github.com> wrote:

> In src/JLD.jl
> <https://github.com/JuliaIO/JLD.jl/pull/72#discussion_r66785962>:
>
> > @@ -1209,7 +1209,7 @@ FileIO.save(f::File{format"JLD"}, value...; kwargs...) = error("Must supply a na
> >  # load with just a filename returns a dictionary containing all the variables
> >  function FileIO.load(f::File{format"JLD"})
> >      jldopen(FileIO.filename(f), "r") do file
> > -        (String => Any)[var => read(file, var) for var in names(file)]
> > +        Dict{String,Any}(var => read(file, var) for var in names(file))
>
> So after seeing the details, this seems to be the problem. This was done
> on mac and julia 0.5(Commit d98f2c0). This seems fine on 0.5, but errors on
> 0.4 inspite of using @compat. I made this change based on the deprecation
> warning,
>
> WARNING: deprecated syntax "[a=>b for (a,b) in c]".
> Use "Dict(a=>b for (a,b) in c)" instead.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaIO/JLD.jl/pull/72/files/cad78486603d894e3710e4cb6726a6f506e2426e#r66785962>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA8SvTloRID2AzclFaVCvGE8bvB5KfsFks5qLVLugaJpZM4Ix3k7>
> .
>